Remove C++11 compatibility from c10::optional

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30919 Remove C++11 compatibility from c10::optional**

deletecode

Differential Revision: [D18869641](https://our.internmc.facebook.com/intern/diff/D18869641/)

smessmer committed:
Remove C++11 compatibility from c10::optional

deletecode

Differential Revision: [D18869641](https://our.internmc.facebook.com/intern/diff/D18869641/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::optional"


deletecode

Differential Revision: [D18869641](https://our.internmc.facebook.com/intern/diff/D18869641/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::optional"


deletecode

Differential Revision: [D18869641](https://our.internmc.facebook.com/intern/diff/D18869641/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::optional"


deletecode

Differential Revision: [D18869641](https://our.internmc.facebook.com/intern/diff/D18869641/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::optional"


deletecode

Differential Revision: [D18869641](https://our.internmc.facebook.com/intern/diff/D18869641/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::optional"


deletecode

Differential Revision: [D18869641](https://our.internmc.facebook.com/intern/diff/D18869641/)

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit 73f69af:
* **1/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=73f69afa19512473d438b9ac91a6e3a0aa7acebe).

### 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3974137) pytorch_linux_xenial_cuda9_cudnn7_py3_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75729453) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75729453))
<details>
<summary>
<code>Dec 17 03:26:22 CMake Error at torch_cuda_generated_THCTensorMathPointwiseByte.cu.o.Release.cmake:281 (message):</code>
</summary>

```
Dec 17 03:26:22 /var/lib/jenkins/workspace/c10/util/ArrayRef.h(278): warning: attribute does not apply to any entity 
Dec 17 03:26:22  
Dec 17 03:26:22 /var/lib/jenkins/workspace/c10/util/Optional.h(456): error: cannot use an entity undefined in device code 
Dec 17 03:26:22  
Dec 17 03:26:22 /var/lib/jenkins/workspace/c10/util/Optional.h(460): error: cannot use an entity undefined in device code 
Dec 17 03:26:22  
Dec 17 03:26:22 /var/lib/jenkins/workspace/c10/util/Optional.h(483): error: identifier "Typeinfo for  ::c10::bad_optional_access" is undefined in device code 
Dec 17 03:26:22  
Dec 17 03:26:22 Error: Internal Compiler Error (codegen): "there was an error in verifying the lgenfe output!" 
Dec 17 03:26:22  
Dec 17 03:26:22 CMake Error at torch_cuda_generated_THCTensorMathPointwiseByte.cu.o.Release.cmake:281 (message): 
Dec 17 03:26:22   Error generating file 
Dec 17 03:26:22   /var/lib/jenkins/workspace/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/generated/./torch_cuda_generated_THCTensorMathPointwiseByte.cu.o 
Dec 17 03:26:22  
Dec 17 03:26:22  
Dec 17 03:26:22 make[2]: *** [caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/generated/torch_cuda_generated_THCTensorMathPointwiseByte.cu.o] Error 1 
Dec 17 03:26:22 caffe2/CMakeFiles/torch_cuda.dir/build.make:210: recipe for target 'caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/generated/torch_cuda_generated_THCTensorMathPointwiseByte.cu.o' failed 
Dec 17 03:26:22 make[2]: *** Waiting for unfinished jobs.... 
Dec 17 03:26:47 /var/lib/jenkins/workspace/c10/util/ArrayRef.h(278): warning: attribute does not apply to any entity 
Dec 17 03:26:47  
Dec 17 03:26:47 /var/lib/jenkins/workspace/c10/util/ArrayRef.h(278): warning: attribute does not apply to any entity 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

smessmer committed:
Update on "Remove C++11 compatibility from c10::optional"


deletecode

Differential Revision: [D18869641](https://our.internmc.facebook.com/intern/diff/D18869641/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::optional"


deletecode

Differential Revision: [D18869641](https://our.internmc.facebook.com/intern/diff/D18869641/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::optional"


deletecode

Differential Revision: [D18869641](https://our.internmc.facebook.com/intern/diff/D18869641/)

[ghstack-poisoned]
